### PR TITLE
chore: fix JSDoc comments in datatype.ts

### PR DESCRIPTION
**Pull request by pkuczynski**
Refs #203

### CODE REVIEW

File: src/datatype.ts [hunk 1]
```diff
@@ -89,11 +109,17 @@ export class Datatype {
   }
 
   /**
-   * Returns a Date object using a random number of milliseconds since 1. Jan 1970 UTC
-   * Caveat: seeding is not working
+   * Returns a Date object using a random number of milliseconds since
+   * the [Unix Epoch](https://en.wikipedia.org/wiki/Unix_time) (1 January 1970 UTC).
```
Comment (Shinigami92) [inline src/datatype.ts:113]: Nice idea :raised_hands:

File: src/datatype.ts [hunk 2]
```diff
@@ -19,10 +19,20 @@ export class Datatype {
   }
 
   /**
-   * Returns a single random number based on a max number or range.
+   * Returns a single random number for the given max value or range and precision.
    *
-   * @method faker.datatype.number
-   * @param options
+   * @param [options] Maximum value or options object.
```
Comment (Shinigami92) [inline src/datatype.ts:24]: I think these brackets are not needed :thinking: 
Could you check how it looks in VSCode?

As far as I know, these are needed in JavaScript, but not in TypeScript

Comment (Shinigami92) [reply src/datatype.ts:24]: an example:

image

seems that it works, but the default values will get eaten :thinking:

Comment (pkuczynski) [reply src/datatype.ts:24]: Brackets means it's an optional param. See https://jsdoc.app/tags-param.html#optional-parameters-and-default-values. I can't check how it looks in VSCode, as I am using JetBrains software... Here it looks like this:

image

**Comment by Shinigami92** [comment]
@ST-DDT could you test out if your api docs generation script can handle this param syntax?

**Comment by ST-DDT** [comment]
@Shinigami92 No it doesn't work.

The typedoc.json does not contain them either.

grafik

File: src/datatype.ts [hunk 3]
```diff
@@ -19,10 +19,20 @@ export class Datatype {
   }
 
   /**
-   * Returns a single random number based on a max number or range.
+   * Returns a single random number for the given max value or range and precision.
```
Comment (ST-DDT) [inline src/datatype.ts:22]:
```suggestion
   * Returns a single random number between zero and the given max value or the given range with the specified precision.
   * The bounds are inclusive.
```

**Comment by pkuczynski** [comment]
> @Shinigami92 No it doesn't work.

There is an issue reported on typedoc and a PR with a fix: https://github.com/TypeStrong/typedoc/issues/567

**Comment by ST-DDT** [comment]
@pkuczynski AFAICT That issue/PR isn't going to be fixed/merged soon and even if it gets merged, it does not solve the problem for us. See https://github.com/TypeStrong/typedoc/pull/1810#pullrequestreview-866938016

So maybe we should remove the `[]` for now?

**Review by Shinigami92 — CHANGES_REQUESTED** [review]
Hey @pkuczynski, I also found out that the usage of square-brackets are not a good idea, cause it could be that JetBrains support it, but Microsoft doesn't support them in VSCode.
So we should try to find a solution to write docs for both worlds.

I also think you are mixing JSDoc and (what I call) TSDoc. It's kinda like JavaDoc vs KotlinDoc, they are similar but not the same.

So what I would suggest is something like
```
/**
 * @param options.min Lower bound for generated number. Defaults to `0`.
 */
```

**Comment by pkuczynski** [comment]
I guess there is no other option than remove those defaults for now, and maybe add them at some point in the future when vscode and typedoc will support them. It's really a shame though, and it displays in the docs much nicer when the `default value` column is provided for a param vs when the information is embedded in the desc.

**Comment by pkuczynski** [comment]
Done. Could you please re-review...?

**Review by Shinigami92 — DISMISSED** [review]

File: src/datatype.ts [hunk 4]
```diff
@@ -64,10 +75,20 @@ export class Datatype {
   }
 
   /**
-   * Returns a single random floating-point number based on a max number or range.
+   * Returns a single random floating-point number for the given precision or range and precision.
+   *
+   * @param options] Precision or options object.
```
Comment (ST-DDT) [inline src/datatype.ts:80]:
```suggestion
   * @param options Precision or options object.
```

File: src/datatype.ts [hunk 5]
```diff
@@ -141,9 +170,10 @@ export class Datatype {
   }
 
   /**
-   * uuid
+   * Returns UUID v4 ([Universally Unique Identifier](https://en.wikipedia.org/wiki/Universally_unique_identifier)).
```
Comment (ST-DDT) [inline src/datatype.ts:173]:
```suggestion
   * Returns a UUID v4 ([Universally Unique Identifier](https://en.wikipedia.org/wiki/Universally_unique_identifier)).
```

File: src/datatype.ts [hunk 6]
```diff
@@ -156,23 +186,28 @@ export class Datatype {
   }
 
   /**
-   * boolean
+   * Returns boolean value true or false.
```
Comment (ST-DDT) [inline src/datatype.ts:189]:
```suggestion
   * Returns the boolean value true or false.
```

File: src/datatype.ts [hunk 7]
```diff
@@ -156,23 +186,28 @@ export class Datatype {
...
+   * @example
+   * faker.datatype.boolean() // false
    */
   boolean(): boolean {
     return !!this.faker.datatype.number(1);
   }
 
   /**
-   * hexaDecimal
+   * Returns [hexadecimal](https://en.wikipedia.org/wiki/Hexadecimal) number.
```
Comment (ST-DDT) [inline src/datatype.ts:199]:
```suggestion
   * Returns a [hexadecimal](https://en.wikipedia.org/wiki/Hexadecimal) number.
```

File: src/datatype.ts [hunk 8]
```diff
@@ -203,14 +238,15 @@ export class Datatype {
   }
 
   /**
-   * Returns json object with 7 pre-defined properties
+   * Returns string representing JSON object with 7 pre-defined properties.
```
Comment (ST-DDT) [inline src/datatype.ts:241]:
```suggestion
   * Returns a string representing JSON object with 7 pre-defined properties.
```

File: src/datatype.ts [hunk 9]
```diff
@@ -221,13 +257,15 @@ export class Datatype {
   }
 
   /**
-   * Returns an array with values generated by faker.datatype.number and faker.datatype.string
+   * Returns array with random strings and numbers.
```
Comment (ST-DDT) [inline src/datatype.ts:260]:
```suggestion
   * Returns an array with random strings and numbers.
```

**Review by ST-DDT — CHANGES_REQUESTED** [review]

File: src/datatype.ts (hunk 4 above)
Comment (pkuczynski) [reply src/datatype.ts]: omg! thanks!

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Review by ST-DDT — APPROVED** [review]

**Pull request merged**